### ISSUE TITLE
Electron 0.31.0 / NaN v2 compatibility

### ISSUE THREAD

**Issue by bpasero** [atom/node-oniguruma#33]
I am trying to build oniguruma against electron 0.31.0 which advanced to io.js 3.1.0. From the release notes I can read that native modules need to update to NaN v2 to be compatible.

Is this planned?

**Comment by paulcbetts** [comment]
This is planned as part of the move to Chrome 44. In the meantime, Electron 0.31.0 should be compatible with the latest nan 1.x release, at least for all the native modules I've been building

**Comment by bpasero** [comment]
@paulcbetts this currently blocks us from adopting Electron 0.31.x. As part of our building process, we configure npm to use the header files from the electron version we use and we always do a full "npm install". It does not really work for us to use precompiled binaries for a previous version.

Any ETA when oniguruma supports building against Electron 0.31.x?

**Comment by paulcbetts** [comment]
@bpasero When you do it? Why don't you guys just port it then contribute back

**Comment by bpasero** [comment]
@paulcbetts no past experience with NaN and I played around with it for 1 hour, hitting some issues. First thing I did was to change the NaN dependency to 2.x (currently 2.0.9). Is there a reason you take the NaN dependency from Atom shell (https://atom.io/download/atom-shell/nan-1.6.1.tgz) and not directly from NPM?

Then when I try to node-gyp against Electron 0.32.x I see errors popping up. Most often it seems to me that code like NanNew<String> now has to be written as Nan::New<v8::String>, but when I do that for the places, I get this error:

../src/onig-scanner.cc:16:21: error: no viable conversion from 'typename imp::Factory<String>::return_t' (aka 'v8::MaybeLocal<v8::String>') to 'Local<v8::String>'
  tpl->SetClassName(Nan::New<v8::String>("OnigScanner"));

If I get more hints how to solve those, I can work on the pull request.
